senders/pushover/pushover.go

```diff
@@ -71,29 +74,85 @@ func (sender *Sender) makePushoverMessage(events moira.NotificationEvents, conta
 	return pushoverMessage
 }
 
-func (sender *Sender) buildMessage(events moira.NotificationEvents, throttled bool) string {
-	var message bytes.Buffer
-	for i, event := range events {
-		if i > printEventsCount-1 {
-			break
+func (sender *Sender) buildMessage(events moira.NotificationEvents, throttled bool, trigger moira.TriggerData) string {
+	var message strings.Builder
+	desc := trigger.Desc
+	htmlDesc := string(blackfriday.Run([]byte(desc)))
+	descLen := len([]rune(desc))
+	htmlDescLen := len([]rune(htmlDesc))
+	charsForHTMLTags := htmlDescLen - descLen
+	var eventsString string
+	for _, event := range events {
+		line := fmt.Sprintf("%s: %s = %s (%s to %s)", event.FormatTimestamp(sender.location), event.Metric, event.GetMetricValue(), event.OldState, event.State)
+		if len(moira.UseString(event.Message)) > 0 {
+			line += fmt.Sprintf(". %s\n", moira.UseString(event.Message))
+		} else {
+			line += "\n"
 		}
-		message.WriteString(fmt.Sprintf("%s: %s = %s (%s to %s)", event.FormatTimestamp(sender.location), event.Metric, event.GetMetricValue(), event.OldState, event.State))
+		eventsString += line
+	}
+	if throttled {
+		eventsString += "\nPlease, fix your system or tune this trigger to generate less events."
+	}
+	eventsStringLen := len([]rune(eventsString))
+	if htmlDescLen+eventsStringLen < msgLimit {
+		// Use both completely
+		message.WriteString(htmlDesc)
+		message.WriteString(eventsString)
+		return message.String()
+	}
+	if htmlDescLen > msgLimit/2 && eventsStringLen > msgLimit/2 {
+		// Trim both desc and events to half the message size each
+		desc = desc[:msgLimit/2-charsForHTMLTags-10] + "...\n"
+		htmlDesc = string(blackfriday.Run([]byte(desc)))
+		eventsString = sender.buildEventsString(events, msgLimit/2, throttled)
+	} else if htmlDescLen > msgLimit/2 {
+		// Trim the desc to the chars left after using the whole events string
+		charsForDesc := msgLimit - eventsStringLen
+		desc = desc[:charsForDesc-charsForHTMLTags-10] + "...\n"
+		htmlDesc = string(blackfriday.Run([]byte(desc)))
+	} else if eventsStringLen > msgLimit/2 {
+		// Trim the events string to the chars left after using the whole desc
+		charsForEvents := msgLimit/2 - htmlDescLen
+		eventsString = sender.buildEventsString(events, charsForEvents, throttled)
+	}
+	message.WriteString(htmlDesc)
+	message.WriteString(eventsString)
+	return message.String()
+}
+func (sender *Sender) buildEventsString(events moira.NotificationEvents, charsForEvents int, throttled bool) string {
+	charsForThrottleMsg := 0
+	if throttled {
+		charsForThrottleMsg = 70
```

Comment:
It's bad to hardcode `70` here. Somebody will change the throttling text and forget to update this value. Better make a constant with the text and use its length.